File: SourceCode/Debug Helpers/open.fda.gov/ApiDalc/DataObjects/RecallSearchResultData.vb
Namespace DataObjects

    ''' <summary>
    ''' Recall Search Results
    ''' </summary>
    ''' <remarks></remarks>
    Public Class RecallSearchResultData

#Region " Public Methods "

        Public Property KeyWord As String
        Public Property Count As Integer
        Public Property Type As String
        Public Property Classification As String
        Public Property ProductDescription As String
        Public Property ReasonForRecall As String

        Public Property Regions As New HashSet(Of String)
        Public Property IsNationWide As Boolean = False

#End Region

    End Class

End Namespace

File: SourceCode/AspCoreVersion/src/ShopAware.Core/DataObjects/RecallSearchResultData.cs
using System.Collections.Generic;

namespace ShopAware.Core
{
    namespace DataObjects
    {
        /// <summary>
        ///     Recall Search Results
        /// </summary>
        /// <remarks></remarks>
        public class RecallSearchResultData
        {
            #region Member Variables

            private bool _IsNationWide = false;

            #endregion

            #region Properties

            public string KeyWord { get; set; }

            public int Count { get; set; }

            public string Type { get; set; }

            public string Classification { get; set; }

            public string ProductDescription { get; set; }

            public string ReasonForRecall { get; set; }

            public HashSet<string> Regions { get; set; }

            public bool IsNationWide
            {
                get
                {
                    return _IsNationWide;
                }
                set
                {
                    _IsNationWide = value;
                }
            }

            #endregion

            #region Constructors

            public RecallSearchResultData()
            {
                // VBConversions Note: Non-static class variable initialization is below.  Class variables cannot be initially assigned non-static values in C#.
                Regions = new HashSet<string>();
            }

            #endregion
        }
    }
}